packages/traceability-schemas/schemas/CommodityType.json

```diff
@@ -0,0 +1,43 @@
+{
+  "$id": "https://w3id.org/traceability/schemas/CommodityType.json",
+  "$schema": "https://json-schema.org/draft-07/schema#",
+  "$comment": "{\"term\": \"CommodityType\", \"@id\": \"https://w3id.org/traceability#CommodityType\"}",
+  "title": "CommodityType",
+  "description": "A type of commodity, used to determine tarifs and required cross border documentation.",
+  "type": "object",
+  "properties": {
+    "@context": {
+      "type": "array"
+    },
+    "type": {
+      "oneOf": [
+        {
+          "type": "string"
+        },
+        {
+          "type": "array"
+        }
+      ]
+    },
+    "commodityDescription": {
+      "$comment": "{\"term\": \"commodityDescription\", \"@id\": \"https://w3id.org/traceability#commodityDescription\"}",
```

Comment:
it is best to avoid creating new terms, so if we can change this `@id` to point to schema.org, gs1 or another more reputable vocabulary we should.

I also like to avoid camel casing when possible, so I would prefer:

```
 "description": {
      "$comment": "{\"term\": \"description\", \"@id\": \"https://schema.org/description\"}",
```

Reply:
This "jam a bunch of structured data into a nominally free-text value, where you must escape many characters along the way" thing is a recipe for disaster.  If it's going to persist a while (as it currently appears), the humans working with these specs will find it easier if some syntactically meaningless whitespace can be used to softly format that free-text, at least in any sample or test data presented in or with these specs, as in ---
```
 "description": {
      "$comment": "{ \"term\":  \"description\", 
                     \"@id\":   \"https://schema.org/description\"
                   }",
```

Reply:
yeah, I agree... the issue is how to extend JSON Schema properly, instead of using the `hack` we invented.

See https://json-schema.org/specification.html

particularly:

- http://json-schema.org/understanding-json-schema/reference/generic.html

We've discussed changing the way this works, do you have any recommendations based on these links?

Reply:
Well, based on [this segment](http://json-schema.org/understanding-json-schema/reference/generic.html#id3) (emphasis below, mine), using `$comment` in this way is effectively forbidden!

>The `$comment` keyword is strictly intended for adding comments to a schema. Its value must always be a string. Unlike the annotations `title`, `description`, and `examples`, **_JSON schema implementations aren’t allowed to attach any meaning or behavior to it whatsoever, and may even strip them at any time._** Therefore, they are useful for leaving notes to future editors of a JSON schema, but should not be used to communicate to users of the schema.

I don't have a solution.  All I can really say is that this is a hazard of constant wheel reinvention (JSON Schema, here) by each new player in the software stack (JSON, here, with which I have many disagreements, not least being the lack of syntax for inline comments which generally should, if not must, be preserved, which really *can* enable self- and/or internally-documented code).

Reply:
from a "JSON Schema" perspective, stripping JSON-LD is fine... the resulting document is still valid JSON Schema without these annotations.

We should explore adding a new property, and setting JSON Schema strict mode to false, that might be more readable.

Reply:
I also wonder if switchin to YAML at the same time, might make this easier to work with.

Reply:
> I also wonder if switching to YAML at the same time, might make this easier to work with.

you know, YAML is not the worst idea in the world for this - might be worth exploring separately